src/dotnet/Telemetry.cs

```diff
@@ -19,6 +20,8 @@ public class Telemetry : ITelemetry
         private Dictionary<string, double> _commonMeasurements = null;
         private Task _trackEventTask = null;
 
+        internal static readonly string TelemetrySentinel  = Path.Combine(ApplicationEnvironment.ApplicationBasePath, $"{Product.Version}.dotnetTelemetry");
```

Comment:
Instead of another sentinel, could we simply reuse the first time sentinel? The first time sentinel gives better guarantees that the telemetry message has been shown and already has guards in place for things like concurrent first invocations.